### ISSUE TITLE
Warn on misleading function call indentation

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#431]
```js
console.log("hello")
(async () => {
  console.log("world")
}());
```

The developer probably expects ASI to happen after `console.log("hello")`. But the code is instead interpreted as a call to the result of `console.log("hello")`:

```js
console.log("hello")(async () => {
  console.log("world")
}());
```

We should emit a warning in the former case.

Criteria (I think):
* Indentation is the same on both lines
* `(` or `[` token on the second line

**Comment by strager** [comment]
Another example:
```js
client.commands = new Collections()
["command", "event"].forEach(addCommand)
```
